docs/I18n_AML.md

```diff
@@ -0,0 +1,229 @@
+add new translations without touching the original - this might be useful for other languages or if a
+project uses its own vocabulary.
+
+#### Option 1: embedded translation
+
+Proposal a)
+
+```
+interaction type select {
+    template = "Select" ${value} "in" ${element}
+    template de = "Wähle" ${value} "für" ${element}
+}
+```
+
+Proposal b)
+
+```
+interaction type select {
+    template = "Select" ${value} "in" ${element}
+          de = "Wähle" ${value} "für" ${element}
+}
+```
+
+:+1: Easy to maintain<br/>
+:+1: Editor support for translations<br/>
+:-1: Harder to add a new language<br/>
+:-1: Not possible to use project specific vocabulary<br/>
+
+#### Option 2: separate translation
+
+First, we would need to name the file. Let's say the name of the AML would be `WebDriverFixture.aml`.
+A properties file could be named `WebDriverFixture_de.properties` and include definitions such as:
+```
+select = "Wähle" ${value} "für" ${element}
+```
+
+:+1: Easier to add a new language<br/>
+:+1: Possible to use project specific vocabulary<br/>
+:-1: Harder to maintain<br/>
+:-1: No editor support in properties file <br/>
+
+#### Translating variables
+
+In the examples above the variables were not translated. If this should be possible we would need to
+have a mapping concept.
+
+For example:
+```
+interaction type select {
+    template = "Select" ${value} "in" ${element}
+    method = MyFixture.selectDropDown(value, element)
+}
+
+// translation (embedded or separate)
+"Für" ${element} "wähle den Wert" ${wert} 
+```
+We would need a way to establish the mapping `wert -> value`.
+
+This could be done by introducing a new syntax such as:
+```
+"Für" ${element} "wähle den Wert" ${wert:value} 
+// or
+"Für" ${element} "wähle den Wert" ${wert->value} 
+```
+
+### Completion proposals
+
+To provide proposals for the user in the editor and the test step selector we somehow need the information
+which language is used within a project. This should not be a user-specific property (see assumptions).
+This could be provided in a project properties file or in the Gradle / Maven build.
+
+## Concept: object names
+
+The name of an object is the identifier which can be used to cross-reference an object.
+In the above example we defined a button with the name `Ok` using:
+```
+// definition in AML
+component DateDialog is Dialog {
+    element Ok is Button
+}
+
+// usage in TCL
+component: DateDialog
+    - Click <Ok>
+```
+
+Currently a name is not translatable and not allowed to contain special characters such as ä, ü, ö or ß.
+Translating the name might be difficult as we would need to adapt the default scoping mechanisms of Xtext.
+We would have not have a single name per object but multiple names and adapt all scoping rules accordingly.
+
+Usually, these objects are defined within a project and not in the reusable parts of the fixture.
```

Comment:
this will also mean that parts that were once decided to be project specific cannot easily be made reusable/public if they make use of elements with special characters. we might make this transition a bit easier by providing refactoring for these element names. thus we could keep the scoping while still being able to push project specific stuff into reusable components.